### PR TITLE
dotCMS/core#15906 SPAEM: Add 2 levels deep to navigation

### PR DESCRIPTION
**Pull request by hmoreras**

### CODE REVIEW

File: src/Components/Nav.js [hunk 1]
```diff
@@ -12,7 +20,7 @@ export default class Nav extends Component {
 
     async componentDidMount() {
```
Comment (alfredo-dotcms) [inline src/Components/Nav.js]: remove `async`

File: src/Components/Nav.js [hunk 2]
```diff
@@ -46,13 +54,25 @@ export default class Nav extends Component {
...
                         {this.state.items.map(item => {
-                            return (
-                                <NavItem key={item.folder}>
-                                    <NavLink tag={Link} active={item.href === window.location.pathname} to={item.href}>
-                                        {item.title}
-                                    </NavLink>
-                                </NavItem>
-                            );
+                            if (item.children.length && item.type === 'folder'){
+                                return(  <UncontrolledDropdown nav inNavbar>
```
Comment (alfredo-dotcms) [inline src/Components/Nav.js]: extra space

File: src/Components/NavOption.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        <NavItem key={item.folder}>
+            <NavLink tag={Link} active={item.href === window.location.pathname} to={item.href}>
+                {item.title}
+            </NavLink>
+        </NavItem>
+    );
+};
+
+NavItem.propTypes = {
+    item: PropTypes.object.isRequired
```
Comment (alfredo-dotcms) [inline src/Components/NavOption.js]: add more detailed prop definition
https://stackoverflow.com/questions/49680907/react-proptype-for-array-and-its-properties

File: src/Components/Nav/Nav.js [hunk 4]
```diff
@@ -45,14 +46,12 @@ export default class Nav extends Component {
...
-                                <NavItem key={item.folder}>
-                                    <NavLink tag={Link} active={item.href === window.location.pathname} to={item.href}>
-                                        {item.title}
-                                    </NavLink>
-                                </NavItem>
-                            );
+                        {this.state.items.map((item, index) => {
+                            if (item.children.length && item.type === 'folder'){
+                                return <NavDropDown key={index} options={item}/>
+                            }else{
```
Comment (fmontes) [inline src/Components/Nav/Nav.js]: prettier? `} else {`

File: src/Components/Nav/Nav.js [hunk 5]
```diff
@@ -45,14 +46,12 @@ export default class Nav extends Component {
...
-                            return (
-                                <NavItem key={item.folder}>
-                                    <NavLink tag={Link} active={item.href === window.location.pathname} to={item.href}>
-                                        {item.title}
-                                    </NavLink>
-                                </NavItem>
-                            );
+                        {this.state.items.map((item, index) => {
+                            if (item.children.length && item.type === 'folder'){
+                                return <NavDropDown key={index} options={item}/>
```
Comment (fmontes) [inline src/Components/Nav/Nav.js]: Each item in the response have a `hash` property that is different for each (like an id) when you have that kind of property in what you render you use those as a key, the index is "ok" tho.

**Pull request merged**